mvc/models/model-binding.rst

```diff
@@ -1,8 +1,92 @@
+
+- ``[BindRequired]``: This attribute forces a model state error if binding cannot occur.
+- ``[BindNever]``: Tells the model binder to never bind to this parameter.
+- ``[FromHeader]``, ``[FromQuery]``, ``[FromRoute]``, ``[FromForm]``: Use these to specify the exact binding source you want to apply.
+- ``[FromServices]``: This attribute uses `dependency injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`_ to bind parameters from services.
+- ``[FromBody]``: Use the configured formatters to bind data from the request body. The formatter is selected based on content type of the request.
+- ``[ModelBinder]``: Used to override the default model binder, binding source and name.
+
+Binding attributes are very helpful tools when you need to override the default behavior of model binding.
+
+Binding formatted data from the request body
+--------------------------------------------
+Request data can come in a variety of formats including JSON, XML and many others. When you use the [FrommBody] attribute to indicate that you want to bind a parameter to data in the request body, MVC uses a configured set of formatters to handle the request data based on its content type. By default MVC includes a ``JsonInputFormatter`` class for handling JSON data, but you can add additional formatters for handling XML and other custom formats. 
+
+.. Note:: The ``JsonInputFormatter`` is based off of Json.NET.
+
+ASP.NET selects input formatters based on the Content-Type header and the type of the parameter, unless it has a model binding attribute specifying otherwise. If you'd like to use XML or another format you must configure it in the Startup.cs file. You may have to install a NuGet package for the formatter you want. Your startup code should look something like this:
+
+.. code:: c#
+
+       public void ConfigureServices(IServiceCollection services)
+       {
+         services.AddMvc(options =>
+         {
+            options.Filters.Add(new FormatFilterAttribute());
+         })
+		 .AddXmlSerializerFormatters()
+         .AddViewLocalization(LanguageViewLocationExpanderFormat.SubFolder);
+       {
+
+Code in the Startup.cs file contains a ``ConfigureServices`` method with a ``services`` argument you can use to build up services for your ASP.NET app. In the sample, we are adding an XML formatter to the list of services MVC will provide for this app.
+
+Resources
+---------
+Below are links to related routing and formatting docs:
```

Comment:
Nuke +Below are links to related routing and formatting docs:
 Redundant info + in the future you'll add more links related to model binding and not routing and formatting.